File: medium/numpairs.rb
File.readlines(ARGV[0]).each do |l|
    l.strip!
    nums, s = l.split(';')
    nums = nums.split(',')
    pairs = []
    nums.each_with_index do |n, i|
        (0..i-1).each do |i2|
            pairs.push("#{nums[i2]},#{n}") if n.to_i + nums[i2].to_i == s.to_i
        end
    end
    if pairs.length > 0
        puts pairs.reverse.join(';')
    else
        puts "NULL"
    end
end

File: medium/numpairs.py
import sys

for l in open(sys.argv[1]):
    l = l.strip()
    nums, s = l.split(';')
    nums = [int(x) for x in nums.split(',')]
    pairs = []
    for i, n in enumerate(nums):
        for n2 in nums[i + 1:]:
            if n + n2 == int(s):
                pairs.append("%d,%d" % (n, n2))

    print ';'.join(pairs) if len(pairs) > 0 else "NULL"
